sandbox/modules/identity/src/main/java/org/opensearch/identity/SecurityFilter.java

```diff
@@ -67,6 +72,26 @@ private <Request extends ActionRequest, Response extends ActionResponse> void ap
         ActionListener<Response> listener,
         ActionFilterChain<Request, Response> chain
     ) {
+        // TODO The section below shows permission check working for index create and index search, but needs
```

Comment:
Do you know what changes will be required here for the plugins and extension functionality? We have the dual meaning of the permissions and extensions since they can both be the subject and have permissions regarding them. Do we need to define any requests for either to move things forward?

Reply:
Extensions can register actions and those actions need to be permissible for cluster admins to be able to permit users to use features of an extension.

Plugins run in the same JVM as the OpenSearch node, but extensions run outside. Extensions interact with the OpenSearch cluster using a rest client to make requests to the cluster. Users have sets of permissions associated with them and extensions may as well, to give cluster admins the ability to further restrict how requests originating from an extension can interact with a cluster. At the very least, an extension making a request to a cluster on behalf of a user should perform the request with the same user permissions + restrictions, restrictions like preventing the ability to query system indices even if a user has access.

I was trying to make an analogy with microsoft word and saving files to the filesystem, but I believe that word is able to save to the same folders as the current logged in user.

Reply:
Oh I see. Thank you for the explanation of what we will need to do moving forward. I have mostly been designing from the standpoint of generic permissions so wanted to see how this would fit with a more user-centric version like you have here. That makes sense though.